metadata-ingestion/src/datahub/ingestion/source/common/resource.py

```diff
@@ -0,0 +1,46 @@
+import json
+from typing import Any, Dict, Optional
+
+import datahub.emitter.mce_builder as builder
+from datahub.metadata.schema_classes import (
+    SerializedValueClass,
+    SerializedValueContentTypeClass,
+)
+from datahub.utilities.str_enum import StrEnum
+
+
+class ResourceType(StrEnum):
+    USER_INFO = "USER_INFO"
+    CHANNEL_INFO = "CHANNEL_INFO"
+    USER_ID_MAPPING = "USER_ID_MAPPING"
+
+
+def generate_user_id_mapping_resource_urn(
```

Comment:
Can get rid of most of these if we use `PlatformResourceKey` and `PlatformResource` from [here](https://github.com/datahub-project/datahub/pull/11541/files#)